src/System.ComponentModel.TypeConverter/src/System/ComponentModel/TypeDescriptor.cs

```diff
@@ -26,8 +26,10 @@ public sealed class TypeDescriptor
         // class load anyway.
         private static readonly WeakHashtable s_providerTable = new WeakHashtable();     // mapping of type or object hash to a provider list
         private static readonly Hashtable s_providerTypeTable = new Hashtable();         // A direct mapping from type to provider.
-        private static volatile Hashtable s_defaultProviders = new Hashtable(); // A table of type -> default provider to track DefaultTypeDescriptionProviderAttributes.
-        private static volatile WeakHashtable s_associationTable;
+
+        // A table of type -> default provider to track DefaultTypeDescriptionProviderAttributes.
+        private static readonly Lazy<Hashtable> s_defaultProvidersLazy = new Lazy<Collections.Hashtable>(() => new Hashtable());
```

Comment:
I just noticed that in the original code, s_defaultProviders is initialized at its declaration site, so it's not actually lazy.  We shouldn't wrap it in Lazy then.  The only thing we should do is change the volatile to readonly.